[OID4VCI] Always Return Array for Credential Responses

forkimenjeckayang opened the pull request:
This PR updates the Keycloak OID4VCI implementation to comply with the latest [OID4VCI specification](https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0-15.html#name-credential-response), ensuring that the `credentials` field in the credential response is always an array (even for a single credential).

- **CredentialResponse Model:**  
  - Updated to ensure `credentials` is always a list of credential objects.
  - Removed/avoided any usage of single credential fields.

- **Issuer Endpoint:**  
  - Ensured that immediate issuance always returns an array of credentials.

### Closses Ticket

- #39283

antikalk committed:
Forward LOGIN_HINT of authentication session with identity-provider-redirector

Fixes keycloak#36396

Signed-off-by: Oliver Cremerius <antikalk@users.noreply.github.com>

douglaspalmer committed:
Access Token IDs have less than 128 bits of entropy

Closes #38663

Signed-off-by: Douglas Palmer <dpalmer@redhat.com>

IngridPuppet commented on services/src/main/java/org/keycloak/protocol/oid4vc/model/CredentialResponse.java:
```diff
@@ -29,18 +32,56 @@
+            this.credential = credential;
+            return this;
+        }
+    }
```
A minor remark that inner classes at usually placed at the bottom of the class.

forkimenjeckayang committed:
Always Return Array of Credentilas for Credential Responses

Closes #39283

Signed-off-by: forkimenjeckayang <forkimenjeckayang@gmail.com>

forkimenjeckayang committed:
Merge branch 'main' into issue-39283

rmartinc committed:
Integrate passkeys with the organization authenticator
Closes #40022

Signed-off-by: rmartinc <rmartinc@redhat.com>

dependabot[bot] committed:
Bump react-hook-form from 7.57.0 to 7.58.1 in /js

Bumps [react-hook-form](https://github.com/react-hook-form/react-hook-form) from 7.57.0 to 7.58.1.
- [Release notes](https://github.com/react-hook-form/react-hook-form/releases)
- [Changelog](https://github.com/react-hook-form/react-hook-form/blob/master/CHANGELOG.md)
- [Commits](https://github.com/react-hook-form/react-hook-form/compare/v7.57.0...v7.58.1)

---
updated-dependencies:
- dependency-name: react-hook-form
  dependency-version: 7.58.1
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump rollup from 4.43.0 to 4.44.0 in /js

Bumps [rollup](https://github.com/rollup/rollup) from 4.43.0 to 4.44.0.
- [Release notes](https://github.com/rollup/rollup/releases)
- [Changelog](https://github.com/rollup/rollup/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rollup/rollup/compare/v4.43.0...v4.44.0)

---
updated-dependencies:
- dependency-name: rollup
  dependency-version: 4.44.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump mocha from 11.7.0 to 11.7.1 in /js

Bumps [mocha](https://github.com/mochajs/mocha) from 11.7.0 to 11.7.1.
- [Release notes](https://github.com/mochajs/mocha/releases)
- [Changelog](https://github.com/mochajs/mocha/blob/main/CHANGELOG.md)
- [Commits](https://github.com/mochajs/mocha/compare/v11.7.0...v11.7.1)

---
updated-dependencies:
- dependency-name: mocha
  dependency-version: 11.7.1
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

mposolda committed:
Update javadoc of java admin-client for Keycloak 26.3
closes #40709

Signed-off-by: mposolda <mposolda@gmail.com>

francis-pouatcha committed:
Merge branch 'main' into issue-39283

ryanemerson committed:
Create clustering test cases for OIDC flows (#40623)

Closes #39965

Signed-off-by: Ryan Emerson <remerson@redhat.com>
Signed-off-by: Michal Hajas <mhajas@redhat.com>
Co-authored-by: Michal Hajas <mhajas@redhat.com>

dependabot[bot] committed:
Bump @types/node from 24.0.3 to 24.0.4 in /js

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 24.0.3 to 24.0.4.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-version: 24.0.4
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

pedroigor committed:
Revert changes from #40497 (#40715)

Signed-off-by: Pedro Igor <pigor.craveiro@gmail.com>

rmartinc committed:
Allow passkeys login when user has no password credential
Closes #40717

Signed-off-by: rmartinc <rmartinc@redhat.com>

dependabot[bot] committed:
Bump vite from 6.3.5 to 7.0.0 in /js

Bumps [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) from 6.3.5 to 7.0.0.
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/main/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/create-vite@7.0.0/packages/vite)

---
updated-dependencies:
- dependency-name: vite
  dependency-version: 7.0.0
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

graziang committed:
Verification of external facebook token via "debug token" endpoint

Closes #40163

Signed-off-by: Giuseppe Graziano <g.graziano94@gmail.com>

Pepo48 committed:
quarkus-next: update Quarkus snapshots url (#40716)

Closes: #40694

Signed-off-by: Peter Zaoral <pzaoral@redhat.com>

shawkins committed:
fix: refining breaking behavior (#40697)

closes: #39063

Signed-off-by: Steve Hawkins <shawkins@redhat.com>

mabartos committed:
Missing Quarkus flag for Syslog counting framing (#40621)

* Missing Quarkus flag for Syslog counting framing

Closes #39893

Signed-off-by: Martin Bartoš <mabartos@redhat.com>

* Remove handling of mapFrom

Signed-off-by: Martin Bartoš <mabartos@redhat.com>

* Add notable change

Signed-off-by: Martin Bartoš <mabartos@redhat.com>

---------

Signed-off-by: Martin Bartoš <mabartos@redhat.com>

pedroigor committed:
Fixing how translatable fields are set and ignoring formatting for root user attributes

Closes #40497

Signed-off-by: Pedro Igor <pigor.craveiro@gmail.com>

pedroigor committed:
Resolve resources with same URI if the permission request is based on URI matching

Closes #40695

Signed-off-by: Pedro Igor <pigor.craveiro@gmail.com>

shawkins committed:
fix: refinements for ipv6 only testing (#40723)

closes: #21277

Signed-off-by: Steve Hawkins <shawkins@redhat.com>

Anchels committed:
Adjusted the comparisons for PersistentClientSessionEntity

Closes #40663

Signed-off-by: Anchels <mishtitov@gmail.com>

rmartinc committed:
Use POST binding for logout when REDIRECT url is not set and forced POST
Closes #40637

Signed-off-by: rmartinc <rmartinc@redhat.com>

weblate committed:
Translations update from Hosted Weblate (#40642)

* Updated translation for Portuguese (Brazil)

Language: pt_BR

Updated translation for Portuguese (Brazil)

Language: pt_BR

Translated using Weblate (Portuguese (Brazil))

Translation: Keycloak/Theme base/admin
Translate-URL: https://hosted.weblate.org/projects/keycloak/theme-baseadmin/pt_BR/

Updated translation for Portuguese (Brazil)

Language: pt_BR

Updated translation for Portuguese (Brazil)

Language: pt_BR

Updated translation for Portuguese (Brazil)

Language: pt_BR

Updated translation for Portuguese (Brazil)

Language: pt_BR

Updated translation for Portuguese (Brazil)

Language: pt_BR

Updated translation for Portuguese (Brazil)

Language: pt_BR

Updated translation for Portuguese (Brazil)

Language: pt_BR

Updated translation for Portuguese (Brazil)

Language: pt_BR

Updated translation for Portuguese (Brazil)

Language: pt_BR

Translated using Weblate (Portuguese (Brazil))

Translation: Keycloak/Theme base/admin
Translate-URL: https://hosted.weblate.org/projects/keycloak/theme-baseadmin/pt_BR/

Updated translation for Portuguese (Brazil)

Language: pt_BR

Updated translation for Portuguese (Brazil)

Language: pt_BR

Updated translation for Portuguese (Brazil)

Language: pt_BR

Updated translation for Portuguese (Brazil)

Language: pt_BR

Co-authored-by: Felipe Zorzo <felipe.b.zorzo@gmail.com>
Co-authored-by: Hosted Weblate <hosted@weblate.org>
Signed-off-by: Felipe Zorzo <felipe.b.zorzo@gmail.com>
Signed-off-by: Hosted Weblate <hosted@weblate.org>

* Updated translation for Japanese

Language: ja

Updated translation for Japanese

Language: ja

Updated translation for Japanese

Language: ja

Updated translation for Japanese

Language: ja

Updated translation for Japanese

Language: ja

Update translation files

Updated by "Squash Git commits" hook in Weblate.

Co-authored-by: Hosted Weblate <hosted@weblate.org>
Co-authored-by: Kohei Tamura <ktamura.biz.80@gmail.com>
Signed-off-by: Hosted Weblate <hosted@weblate.org>
Signed-off-by: Kohei Tamura <ktamura.biz.80@gmail.com>

* Updated translation for Catalan

Language: ca

Updated translation for Catalan

Language: ca

Updated translation for Catalan

Language: ca

Updated translation for Catalan

Language: ca

Updated translation for Catalan

Language: ca

Updated translation for Catalan

Language: ca

Co-authored-by: Ecron <ecron_89@hotmail.com>
Co-authored-by: Hosted Weblate <hosted@weblate.org>
Co-authored-by: Jordi Mallach <jordi@mallach.net>
Signed-off-by: Ecron <ecron_89@hotmail.com>
Signed-off-by: Hosted Weblate <hosted@weblate.org>
Signed-off-by: Jordi Mallach <jordi@mallach.net>

* Updated translation for French

Language: fr

Updated translation for French

Language: fr

Co-authored-by: Hosted Weblate <hosted@weblate.org>
Co-authored-by: Sylvain Pichon <service@spichon.fr>
Signed-off-by: Hosted Weblate <hosted@weblate.org>
Signed-off-by: Sylvain Pichon <service@spichon.fr>

---------

Signed-off-by: Felipe Zorzo <felipe.b.zorzo@gmail.com>
Signed-off-by: Hosted Weblate <hosted@weblate.org>
Signed-off-by: Kohei Tamura <ktamura.biz.80@gmail.com>
Signed-off-by: Ecron <ecron_89@hotmail.com>
Signed-off-by: Jordi Mallach <jordi@mallach.net>
Signed-off-by: Sylvain Pichon <service@spichon.fr>
Co-authored-by: Felipe Zorzo <felipe.b.zorzo@gmail.com>
Co-authored-by: Kohei Tamura <ktamura.biz.80@gmail.com>
Co-authored-by: Ecron <ecron_89@hotmail.com>
Co-authored-by: Jordi Mallach <jordi@mallach.net>
Co-authored-by: Sylvain Pichon <service@spichon.fr>

lhanusov committed:
[Test framework] ImpersonationTest migration

Part of #34494

Signed-off-by: Lukas Hanusovsky <lhanusov@redhat.com>

ahus1 committed:
Set a unique fragment key to please React

Closes #40598

Signed-off-by: Alexander Schwartz <aschwart@redhat.com>

mhajas committed:
Make rolling-updates-v2 preview feature (#40732)


Closes #38883
Signed-off-by: Michal Hajas <mhajas@redhat.com>

dependabot[bot] committed:
Bump eslint-plugin-prettier from 5.5.0 to 5.5.1 in /js (#40725)

Bumps [eslint-plugin-prettier](https://github.com/prettier/eslint-plugin-prettier) from 5.5.0 to 5.5.1.
- [Release notes](https://github.com/prettier/eslint-plugin-prettier/releases)
- [Changelog](https://github.com/prettier/eslint-plugin-prettier/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prettier/eslint-plugin-prettier/compare/v5.5.0...v5.5.1)

---
updated-dependencies:
- dependency-name: eslint-plugin-prettier
  dependency-version: 5.5.1
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @eslint/compat from 1.3.0 to 1.3.1 in /js (#40727)

Bumps [@eslint/compat](https://github.com/eslint/rewrite/tree/HEAD/packages/compat) from 1.3.0 to 1.3.1.
- [Release notes](https://github.com/eslint/rewrite/releases)
- [Changelog](https://github.com/eslint/rewrite/blob/main/packages/compat/CHANGELOG.md)
- [Commits](https://github.com/eslint/rewrite/commits/compat-v1.3.1/packages/compat)

---
updated-dependencies:
- dependency-name: "@eslint/compat"
  dependency-version: 1.3.1
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

petrov9 committed:
Support all i18n keys for messages_ru.properties

Now all messages_ru.properties file is a full copy of a related messages_en.properties file but with translated values

Closes #40336

Signed-off-by: Anton Petrov <petrov9810@gmail.com>
Signed-off-by: Alexander Schwartz <alexander.schwartz@gmx.net>
Co-authored-by: Alexander Schwartz <alexander.schwartz@gmx.net>

ahus1 committed:
Move section in the upgrading guide

Closes #39063

Signed-off-by: Alexander Schwartz <aschwart@redhat.com>

weblate committed:
Translations update from Hosted Weblate (#40745)

* Updated translation for German

Language: de

Updated translation for German

Language: de

Updated translation for German

Language: de

Updated translation for German

Language: de

Updated translation for German

Language: de

Translated using Weblate (German)

Translation: Keycloak/Theme base/admin
Translate-URL: https://hosted.weblate.org/projects/keycloak/theme-baseadmin/de/

Co-authored-by: Alexander Schwartz <alexander.schwartz@gmx.net>
Co-authored-by: Hosted Weblate <hosted@weblate.org>
Signed-off-by: Alexander Schwartz <alexander.schwartz@gmx.net>
Signed-off-by: Hosted Weblate <hosted@weblate.org>

* Updated translation for Russian

Language: ru

Updated translation for Russian

Language: ru

Updated translation for Russian

Language: ru

Updated translation for Russian

Language: ru

Translated using Weblate (Russian)

Translation: Keycloak/Theme base/admin
Translate-URL: https://hosted.weblate.org/projects/keycloak/theme-baseadmin/ru/

Updated translation for Russian

Language: ru

Updated translation for Russian

Language: ru

Updated translation for Russian

Language: ru

Translated using Weblate (Russian)

Translation: Keycloak/Theme base/admin
Translate-URL: https://hosted.weblate.org/projects/keycloak/theme-baseadmin/ru/

Translated using Weblate (Russian)

Translation: Keycloak/Theme base/admin
Translate-URL: https://hosted.weblate.org/projects/keycloak/theme-baseadmin/ru/

Updated translation for Russian

Language: ru

Updated translation for Russian

Language: ru

Updated translation for Russian

Language: ru

Updated translation for Russian

Language: ru

Updated translation for Russian

Language: ru

Update translation files

Updated by "Cleanup translation files" hook in Weblate.

Co-authored-by: Alexander Schwartz <alexander.schwartz@gmx.net>
Co-authored-by: Anton Petrov <petrov9810@gmail.com>
Co-authored-by: Hosted Weblate <hosted@weblate.org>
Signed-off-by: Alexander Schwartz <alexander.schwartz@gmx.net>
Signed-off-by: Anton Petrov <petrov9810@gmail.com>
Signed-off-by: Hosted Weblate <hosted@weblate.org>

* Updated translation for Japanese

Language: ja

Updated translation for Japanese

Language: ja

Updated translation for Japanese

Language: ja

Updated translation for Japanese

Language: ja

Translated using Weblate (Japanese)

Translation: Keycloak/Theme base/admin
Translate-URL: https://hosted.weblate.org/projects/keycloak/theme-baseadmin/ja/

Updated translation for Japanese

Language: ja

Co-authored-by: Hosted Weblate <hosted@weblate.org>
Co-authored-by: Kohei Tamura <ktamura.biz.80@gmail.com>
Signed-off-by: Hosted Weblate <hosted@weblate.org>
Signed-off-by: Kohei Tamura <ktamura.biz.80@gmail.com>

* Updated translation for Catalan

Language: ca

Updated translation for Catalan

Language: ca

Updated translation for Catalan

Language: ca

Co-authored-by: Hosted Weblate <hosted@weblate.org>
Co-authored-by: Jordi Mallach <jordi@mallach.net>
Signed-off-by: Hosted Weblate <hosted@weblate.org>
Signed-off-by: Jordi Mallach <jordi@mallach.net>

* Updated translation for Spanish

Language: es

Updated translation for Spanish

Language: es

Updated translation for Spanish

Language: es

Updated translation for Spanish

Language: es

Co-authored-by: Hernan Lopez <hernan.lopez@gmail.com>
Co-authored-by: Hosted Weblate <hosted@weblate.org>
Signed-off-by: Hernan Lopez <hernan.lopez@gmail.com>
Signed-off-by: Hosted Weblate <hosted@weblate.org>

* Updated translation for French

Language: fr

Updated translation for French

Language: fr

Updated translation for French

Language: fr

Updated translation for French

Language: fr

Updated translation for French

Language: fr

Co-authored-by: Alexander Schwartz <alexander.schwartz@gmx.net>
Co-authored-by: Hosted Weblate <hosted@weblate.org>
Co-authored-by: Sylvain Pichon <service@spichon.fr>
Signed-off-by: Alexander Schwartz <alexander.schwartz@gmx.net>
Signed-off-by: Hosted Weblate <hosted@weblate.org>
Signed-off-by: Sylvain Pichon <service@spichon.fr>

---------

Signed-off-by: Alexander Schwartz <alexander.schwartz@gmx.net>
Signed-off-by: Hosted Weblate <hosted@weblate.org>
Signed-off-by: Anton Petrov <petrov9810@gmail.com>
Signed-off-by: Kohei Tamura <ktamura.biz.80@gmail.com>
Signed-off-by: Jordi Mallach <jordi@mallach.net>
Signed-off-by: Hernan Lopez <hernan.lopez@gmail.com>
Signed-off-by: Sylvain Pichon <service@spichon.fr>
Co-authored-by: Alexander Schwartz <alexander.schwartz@gmx.net>
Co-authored-by: Anton Petrov <petrov9810@gmail.com>
Co-authored-by: Kohei Tamura <ktamura.biz.80@gmail.com>
Co-authored-by: Jordi Mallach <jordi@mallach.net>
Co-authored-by: Hernan Lopez <hernan.lopez@gmail.com>
Co-authored-by: Sylvain Pichon <service@spichon.fr>

rmartinc committed:
Documentation changes for Passkeys (#40728)

Closes #40705

Signed-off-by: rmartinc <rmartinc@redhat.com>


Co-authored-by: Alexander Schwartz <alexander.schwartz@gmx.net>
Signed-off-by: Ricardo Martin <rmartinc@redhat.com>

pedroigor committed:
Do not show update email link if the email attribute is not writable

Closes #39669

Signed-off-by: Pedro Igor <pigor.craveiro@gmail.com>

dasniko committed:
make abstract class AbstractUserRoleMappingMapper public (was package-private)

closes #40765

Signed-off-by: Niko Köbler <niko@n-k.de>

weblate committed:
Translations update from Hosted Weblate (#40771)

* Updated translation for German

Language: de

Co-authored-by: Hosted Weblate <hosted@weblate.org>
Co-authored-by: Robin <39960884+robson90@users.noreply.github.com>
Signed-off-by: Hosted Weblate <hosted@weblate.org>
Signed-off-by: Robin <39960884+robson90@users.noreply.github.com>

* Updated translation for Japanese

Language: ja

Updated translation for Japanese

Language: ja

Updated translation for Japanese

Language: ja

Co-authored-by: Hosted Weblate <hosted@weblate.org>
Co-authored-by: Kohei Tamura <ktamura.biz.80@gmail.com>
Signed-off-by: Hosted Weblate <hosted@weblate.org>
Signed-off-by: Kohei Tamura <ktamura.biz.80@gmail.com>

---------

Signed-off-by: Hosted Weblate <hosted@weblate.org>
Signed-off-by: Robin <39960884+robson90@users.noreply.github.com>
Signed-off-by: Kohei Tamura <ktamura.biz.80@gmail.com>
Co-authored-by: Robin <39960884+robson90@users.noreply.github.com>
Co-authored-by: Kohei Tamura <ktamura.biz.80@gmail.com>

weblate committed:
Updated translation for Russian (#40774)

Language: ru

Updated translation for Russian

Language: ru

Signed-off-by: Anton Petrov <petrov9810@gmail.com>
Signed-off-by: Hosted Weblate <hosted@weblate.org>
Co-authored-by: Anton Petrov <petrov9810@gmail.com>

posymeon committed:
Added missing translations for greek translation in the login page

Closes #40632

Signed-off-by: Polychronis Symeonidis <posymeon@gmail.com>
Co-authored-by: Polychronis Symeonidis <posymeon@gmail.com>

forkimenjeckayang committed:
Merge branch 'main' into issue-39283

forkimenjeckayang committed:
Merge branch 'main' into issue-39283
